File: 30 Days of Code/PHP/Day 20 - Sorting.php
<?php



$n = intval(trim(fgets(STDIN)));

$a_temp = rtrim(fgets(STDIN));

$a = array_map('intval', preg_split('/ /', $a_temp, -1, PREG_SPLIT_NO_EMPTY));

$swap = 0;
// Write your code here
for ($i = 0; $i < $n; $i++) {
    // Track number of elements swapped during a single array traversal
    $numberOfSwaps = 0;
    
    for ($j = 0; $j < $n - 1; $j++) {
        // Swap adjacent elements if they are in decreasing order
        if ($a[$j] > $a[$j + 1]) {
            $temp = $a[$j];
            $a[$j] = $a[$j + 1];
            $a[$j + 1] = $temp;
            $numberOfSwaps++;
        }
    }
    $swap += $numberOfSwaps;
    // If no elements were swapped during a traversal, array is sorted
    if ($numberOfSwaps == 0) {
        break;
    }
}
echo("Array is sorted in ". $swap ." swaps.\n");
echo("First Element: ".$a[0]."\n");
echo("Last Element: ".$a[$n-1]."\n");
?>

File: 30 Days of Code/Python/Day 20 - Sorting.py
#!/bin/python3

import math
import os
import random
import re
import sys



if __name__ == '__main__':
    n = int(input().strip())

    a = list(map(int, input().rstrip().split()))

    # Write your code here
    swap = 0
    for i in range(n):
        numberOfSwaps = 0
    
        for j in range(n-1):
            # Swap adjacent elements if they are in decreasing order
            if a[j] > a[j + 1]:
                a[j], a[j + 1] = a[j + 1], a[j]
                numberOfSwaps += 1
        swap += numberOfSwaps
        # If no elements were swapped during a traversal, array is sorted
        if (numberOfSwaps == 0):
            break
    
    print("Array is sorted in", swap, "swaps.")
    print("First Element:",a[0])
    print("Last Element:",a[n-1])

